.github/actions/setup-build/action.yml

```diff
@@ -0,0 +1,233 @@
+    with:
+      maven-version: ${{ inputs.maven-version }}
+      set-mvnw: true
+  # - name: Configure Maven cache
+  #   uses: ./.github/actions/setup-maven-cache
+  #   with:
+  #     maven-cache-key-modifier: ${{ inputs.maven-cache-key-modifier }}
+  # Camunda Nexus and extra mirrors/servers must be merged on ids (extra values have precedence).
+  - name: Get Maven mirrors and servers to use
+    id: maven
+    env:
+      # Camunda Nexus is enabled only if repo is owned by Camunda
+      CAMUNDA_NEXUS_ENABLED: >-
+        ${{ 
+          steps.checks.outputs.repo-owned-by-camunda == 'true' &&
+          inputs.camunda-nexus == 'true'
+        }}
+      CAMUNDA_NEXUS_MAVEN_MIRROR: >-
+        [{
+          "id": "camunda-nexus",
+          "name": "Camunda Nexus",
+          "url": "https://repository.nexus.camunda.cloud/content/groups/internal/",
+          "mirrorOf": "zeebe,zeebe-snapshots"
+        }]
+      CAMUNDA_NEXUS_MAVEN_SERVER: >-
+        [{
+          "id": "camunda-nexus",
+          "username": "${{ steps.secrets.outputs.camunda-nexus-username }}",
+          "password": "${{ steps.secrets.outputs.camunda-nexus-password }}"
+        }]
+      EXTRA_MAVEN_MIRRORS: ${{ inputs.maven-mirrors }}
+      EXTRA_MAVEN_SERVERS: ${{ inputs.maven-servers }}
+    run: |
+      # If Camunda Nexus is disabled, reset default mirrors/servers to empty lists
+      if [ "${CAMUNDA_NEXUS_ENABLED}" != "true" ]; then
+        CAMUNDA_NEXUS_MAVEN_MIRROR="[]"
+        CAMUNDA_NEXUS_MAVEN_SERVER="[]"
+      fi
+
+      # Use jq 'add' operator to merge grouped sublists
+      mirrors=$(\
+        jq -n \
+        --argjson nexus_mirror "${CAMUNDA_NEXUS_MAVEN_MIRROR}" \
+        --argjson maven_mirrors "${EXTRA_MAVEN_MIRRORS}" \
+        '[ $nexus_mirror + $maven_mirrors | group_by(.id)[] | add ]'
+      )
+      servers=$(\
+        jq -n \
+        --argjson nexus_server "${CAMUNDA_NEXUS_MAVEN_SERVER}" \
+        --argjson maven_servers "${EXTRA_MAVEN_SERVERS}" \
+        '[ $nexus_server + $maven_servers | group_by(.id)[] | add ]'
+      )
+      
+      echo mirrors=$mirrors >> $GITHUB_OUTPUT
+      echo servers=$servers >> $GITHUB_OUTPUT
+    shell: bash
+  - name: Update Maven settings.xml
+    uses: s4u/maven-settings-action@v3.0.0
+    with:
+      githubServer: false
+      servers: ${{ steps.maven.outputs.servers }}
+      mirrors: ${{ steps.maven.outputs.mirrors }}
+  - name: Set Time Zone
+    if: inputs.time-zone != '' && runner.os == 'Linux'
+    env:
+      TZ_IDENTIFIER: ${{ inputs.time-zone }}
+    run: |
+      # tzdata is not installed in self-hosted runners
+      sudo apt-get update && sudo apt-get install -y tzdata
+      echo ${TZ_IDENTIFIER} | sudo tee /etc/timezone 
+      sudo rm -rf /etc/localtime
+      sudo ln -s /usr/share/zoneinfo/${TZ_IDENTIFIER} /etc/localtime
+    shell: bash
+  - name: Login to Docker Hub
+    # Camunda Nexus is enabled only if repo is owned by Camunda (not for external forks)
```

Comment:
❓ "Camunda Nexus" should be "DockerHub" I guess

Reply:
Nvm, addressed in follow-up PR https://github.com/camunda/camunda/pull/24409/commits/a7435821ad4261309351823d958ad09793398015?diff=unified&w=0

Reply:
Yes indeed :)